Apps/WebClient/src/ClientApp/src/components/services/OrganDonorDetailsCard.vue

```diff
@@ -0,0 +1,183 @@
+﻿<script lang="ts">
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faCircleInfo, faDownload } from "@fortawesome/free-solid-svg-icons";
+import saveAs from "file-saver";
+import Vue from "vue";
+import { Component, Prop, Ref } from "vue-property-decorator";
+import { Action, Getter } from "vuex-class";
+
+import MessageModalComponent from "@/components/modal/MessageModalComponent.vue";
+import PatientData, {
+    HealthOptionTypes,
+    OrganDonorRegistrationData,
+    PatientDataFile,
+    PatientHealthOptions,
+} from "@/models/patientData";
+import container from "@/plugins/container";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import { ILogger } from "@/services/interfaces";
+import SnowPlow from "@/utility/snowPlow";
+
+library.add(faCircleInfo, faDownload);
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const options: any = {
+    components: { MessageModalComponent },
+};
+
+@Component(options)
+export default class OrganDonorDetailsCard extends Vue {
+    @Prop({ required: true })
+    hdid!: string;
+
+    @Getter("patientData", { namespace: "patientData" })
+    getPatientData!: (hdid: string) => PatientData;
+
+    @Action("retrievePatientDataFile", { namespace: "patientData" })
+    getPatientDataFile!: (params: {
+        fileId: string;
+        hdid: string;
+    }) => Promise<PatientDataFile>;
+
+    @Ref("sensitiveDocumentModal")
+    readonly sensitiveDocumentModal!: MessageModalComponent;
+
+    logger!: ILogger;
+
+    get registrationData(): OrganDonorRegistrationData | undefined {
+        return this.patientData.items.find(
+            (ho: PatientHealthOptions) =>
+                ho.type === HealthOptionTypes.OrganDonorRegistrationData
+        ) as OrganDonorRegistrationData;
+    }
+
+    get patientData(): PatientData {
+        return this.getPatientData(this.hdid);
+    }
+
+    private created(): void {
+        this.logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+    }
+
+    private showConfirmationModal(): void {
+        this.sensitiveDocumentModal.showModal();
+    }
+
+    private getDecisionFile(): void {
+        if (this.registrationData && this.registrationData.registrationFileId) {
+            SnowPlow.trackEvent({
+                action: "download_report",
+                text: "Organ Donor",
+            });
+            this.getPatientDataFile({
+                fileId: this.registrationData.registrationFileId,
+                hdid: this.hdid,
+            })
+                .then(
+                    (patientFile: PatientDataFile) =>
+                        new Blob(patientFile.content, {
+                            type: patientFile.contentType,
+                        })
+                )
+                .then((blob) => saveAs(blob, `Organ_Donor_Registration.pdf`))
```

Comment:
How does this compare to the .fetch() and .blob() used elsewhere?

Reply:
https://github.com/bcgov/healthgateway/blob/664594d16011d0134c67aff02261214e560de22a/Apps/WebClient/src/ClientApp/src/components/timeline/entryCard/ClinicalDocumentTimelineComponent.vue#L94

The other data was returned ass encoded media and a BASE64 data string was built with the properties. so As this is a byte I should in theory be able to build out the blob directly